### PR TITLE
bithdtv: improve imdb search. resolves #7596

### LINKED ISSUE #7596: [BIT-HDTV] Configure IMDb ID search to look in both 'URL' and 'Title & Desc.' fields
Is it possible to configure Jackett so that when the IMDb ID is used to search BIT-HDTV, it searches both the URL and Title & Desc.?

image
image

That's particularly useful for TV season packs (e.g. tt0068098) and movie packs (e.g. tt0364569) where the uploader may not enter the IMDb ID in the URL field, but mentions it in the torrent description.

### PR DESCRIPTION
**Pull request by cadatoiva**
Fixes #7596

### CODE REVIEW

File: src/Jackett.Common/Indexers/BitHdtv.cs [hunk 1]
```diff
@@ -123,95 +115,96 @@ public override async Task<IndexerConfigurationStatus> ApplyConfiguration(JToken
...
-            if (!string.IsNullOrWhiteSpace(query.ImdbID))
+            var results = new List<WebClientStringResult>();
+            var search = new UriBuilder(SearchUrl);
+            if (query.IsImdbQuery)
             {
                 qc.Add("search", query.ImdbID);
-                qc.Add("options", "4");
+                qc.Add("options", "4"); //Search URL field for IMDB link
+                search.Query = qc.GetQueryString();
+                results.Add(await RequestStringWithCookiesAndRetry(search.ToString()));
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/BitHdtv.cs]: I like more the old school method `SearchUrl + qc.GetQueryString()` but up to you.
I tested the tracker and it's working.

**Comment by ngosang** [comment]
@cadatoiva please, change the title, we agree on `bithdtv: improve imdb search. resolves #7596` until the bot can handle it

**Comment by ngosang** [comment]
@cadatoiva is this ready?

**Comment by cadatoiva** [comment]
This is ready as is for the IMDB search. It's just missing the Category fix you were asking for in another issue, hence the WIP.

**Pull request merged**